roles/certs/acme-host-cert/tasks/manage-dns-records.yml

```diff
@@ -0,0 +1,26 @@
+---
+
+- block:
+  - name: "Manage dns TXT records to be validated by ACME ({{ dns_state }})"
+    set_fact:
+      certificate_dns_entries: "{{ certificate_dns_entries | default([]) +
+                                    [{
+                                      'type': 'TXT',
+                                      'ttl': '30',
+                                      'state': dns_state,
+                                      'record': item.0,
+                                      'value': item.1
+                                    }]
+                                }}"
+    loop: "{{ acme_certificate_output.challenge_data_dns | dictsort }}"
+
+  - name: "Call 'manage-dns-records' to update DNS server(s)"
+    import_role:
+      name: "../../../dns/manage-dns-records"
```

Comment:
Tried running this on both OSX and Fedora 29 with Ansible 2.7 and Python 2.7.15. Ansible was complaining about not being able to find a role called `../../../dns/manage-dns-records`. Changing it to just search for a role called `dns/manage-dns-records` fixed this problem. Sounds like you and Ben didn't have any issue with this though, so may need to dig into whether this is a "my machine" thing.

Reply:
OK, good data points. I'll take a closer look at the search path for the role to ensure that's not a problem. 

re: `prereq` - yes, agree that we should have a way to install these, but I don't think it's a good idea to have them be "always" part of the the execution. Reason being that you should be able to run through this scenario where you may not have install permissions, and hence adding such as prereq would make the implementation fail. However, we can look to use the `always` and `never` tags and only do it if requested + maybe add another set of tasks to detect if the prereqs are installed and if not just error out with an appropriate message.

Reply:
Hey, I actually did have to change the the role path to `dns/manage-dns-records` as well. It looks like ansible searches for roles here from the context of the playbook directory. Apologies for failing to capture this.